resources/assets/coffee/react/beatmapset-page/main.coffee

```diff
@@ -25,25 +25,39 @@ BeatmapsetPage.Main = React.createClass
     optionsHash = BeatmapsetPageHash.parse location.hash
     @initialPage = optionsHash.page
 
-    beatmaps = _.keyBy @props.beatmapset.beatmaps.data, (o) -> o.id
+    beatmaps = _.concat @props.beatmapset.beatmaps.data, @props.beatmapset.converts.data
+    beatmaps = _.sortBy beatmaps, ['convert', 'difficulty_rating']
 
-    currentBeatmapId =
-      if optionsHash.beatmapId? && beatmaps[optionsHash.beatmapId]?
-        optionsHash.beatmapId
+    # group beatmaps by playmode and then by beatmap id
+    beatmaps = _.groupBy beatmaps, 'mode'
+    # contains the beatmap ids in their appropriate order
+    beatmapList = {}
```

Comment:
```
for key, val of beatmaps
      beatmaps[key] = _.keyBy val, 'id'
      beatmapList[key] = _.map val, 'id'
```

`beatmaps` is the beatmap objects grouped by modes, so that converts (having the same id in each mode) can be retrieved by the mode + id pair.

`beatmapList` (don't mind my terrible naming) is an array of beatmap ids in the proper, sorted order.